addons/mail/static/src/core/web/activity.js

```diff
@@ -128,4 +128,29 @@ export class Activity extends Component {
     get thread() {
         return this.threadService.getThread(this.props.data.res_model, this.props.data.res_id);
     }
+
+    /**
+     * @param {MouseEvent} ev
+     */
+    async onClick(ev) {
+        const model = ev.target.dataset.oeModel;
+        const id = Number(ev.target.dataset.oeId);
+        if (ev.target.tagName === "A") {
+            if (model && id) {
+                ev.preventDefault();
+                await this.env.services.action.doAction({
+                    type: "ir.actions.act_window",
+                    res_model: model,
+                    views: [[false, "form"]],
+                    res_id: id,
+                });
+                if (!this.env.isSmall) {
+                    this.threadService.open(this.thread, true, {
+                        autofocus: false
+                    });
+                }
+            }
+            return;
+        }
```

Comment:
Wouldn't it be possible to have more common code with the similar processing done in messages, rather than duplicating it?